src/scripts/main.js

```diff
@@ -1,3 +1,253 @@
+      <h2 class="title">${title}</h2>
+      <p>${description}</p>
+  </div>
+  `);
+
+  setTimeout(() => {
+    body.removeChild(document.querySelector('.notification'));
+  }, 2000);
+}
+
+function checkFormData(key, value) {
+  if (!value) {
+    pushNotification('An error occured', 'Empty field value', 'error');
+
+    return false;
+  }
+
+  if (key === 'name' && value.length < 4) {
+    pushNotification('An error occured', 'Invalid name value', 'error');
+
+    return false;
+  }
+
+  if (key === 'age' && ((+value < 18 || +value > 90) || isNaN(+value))) {
+    pushNotification('An error occured', 'Invalid age value', 'error');
+
+    return false;
+  }
+
+  return true;
+}
+
+tHead.addEventListener('click', (e) => {
+  const index = e.target.cellIndex;
+  const isSorted = e.target.matches('.sorted');
+  const isSalary = e.target.innerText === 'Salary';
+
+  [...tBody.children]
+    .sort((a, b) => {
+      const aValue = isSalary
+        ? +(a.children[index].innerText.replaceAll(/\$|,/g, ''))
+        : a.children[index].innerText;
+      const bValue = isSalary
+        ? +(b.children[index].innerText.replaceAll(/\$|,/g, ''))
+        : b.children[index].innerText;
+
+      return aValue > bValue
+        ? isSorted ? -1 : 1
+        : aValue < bValue
+          ? isSorted ? 1 : -1
+          : 0;
+    })
+    .forEach(item => tBody.append(item));
+
+  e.target.classList.toggle('sorted', !isSorted);
+
+  [...tHead.firstElementChild.children].forEach(item => {
+    if (item !== e.target) {
+      item.classList.remove('sorted');
+    }
+  });
+});
+
+tBody.addEventListener('click', (e) => {
+  if (e.target.closest('tr').matches('.active')) {
+    return;
+  }
+
+  [...tBody.children].forEach(item => {
+    item.classList.remove('active');
+  });
+
+  e.target.closest('tr').classList.toggle('active');
+});
+
+body.querySelector('.new-employee-form').addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const data = new FormData(e.target);
+
+  const newNode = document.createElement('tr');
+
+  for (const [key, value] of data) {
+    if (!checkFormData(key, value)) {
+      return;
+    }
+
+    if (key === 'salary') {
+      newNode.insertAdjacentHTML('beforeend', `
+        <td>$${Intl.NumberFormat('en-US').format(+value)}</td>
+      `);
+    } else {
+      newNode.insertAdjacentHTML('beforeend', `<td>${value}</td>`);
+    }
+  }
+
+  tBody.append(newNode);
+  e.target.reset();
+  pushNotification('Success', 'New data added', 'success');
+});
+
+tBody.addEventListener('dblclick', ev => {
+  const targeted = ev.target;
+
+  const originalText = targeted.textContent;
+
+  const input = document.createElement('input');
+
+  input.className = 'cell-input';
+  input.name = 'text';
+  input.type = 'text';
+
+  targeted.textContent = '';
+
+  targeted.append(input);
+  input.focus();
+
+  const textInput = document.querySelector('.cell-input');
+
+  input.addEventListener('keydown', e => {
```

Comment:
will be awesome to find the way how you can optimize the handler for this event, right now it's not clear and hard to read your code